### PR TITLE
fix: payload transformations respects log level checks

### LINKED ISSUE #65: Encryption and Compression is done before checking log level
In `CustomLoggerOperation`, `transformPayload` is being called before the log operation is being called, so the payload is being compressed and encrypted before checking if the message should be logged.

The logic for encrypting and compressing needs to be moved so the log level is checked before the payload is encrypted or compressed.

### PR DESCRIPTION
**Pull request by manikmagar**
Closes #65

### CODE REVIEW

File: src/test/java/com/avioconsulting/mule/logger/internal/CustomLoggerOperationTest.java [hunk 1]
```diff
@@ -62,12 +63,12 @@ public void log_compressedPayloadIsSet_test() throws MuleException {
     CustomLoggerOperation spyCustomLoggerOperation = spy(customLoggerOperation);
     LogProperties logProperties = new LogProperties();
     LogProperties spyLogProperties = spy(logProperties);
-    ExtensionsClient extensionsClient = mock(ExtensionsClient.class);
-    spyCustomLoggerOperation.extensionsClient = extensionsClient;
     ParameterResolver<String> resolver = mock(ParameterResolver.class);
 
     // create mock for custom logger and mock log() method so it's never called
     CustomLoggerConfiguration loggerConfig = mock(CustomLoggerConfiguration.class);
+    ExtensionsClient extensionsClient = mock(ExtensionsClient.class);
```
[the remark this answers is not in the document]
Comment (manikmagar) [reply src/test/java/com/avioconsulting/mule/logger/internal/CustomLoggerOperationTest.java:70]: no, we are not. Right now, they are just adjusted to new code changes. Those definitely need a review and improvements. Will be keeping this draft until we get it improved.

[the remark this answers is not in the document]
Comment (adesjardin) [reply src/test/java/com/avioconsulting/mule/logger/internal/CustomLoggerOperationTest.java:70]: OK, just making sure I wasn't missing something in there.    We probably need to get the fix out sooner than later so manual testing in studio may be fine for the short term as well.

File: src/main/java/com/avioconsulting/mule/logger/internal/utils/PayloadTransformer.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      StreamingHelper streamingHelper, String payloadString) {
+    Objects.requireNonNull(streamingHelper, "StreamingHelper cannot be null");
+    if (payloadString == null)
+      return payloadString;
+    Compressor compressor = loggerConfig.getCompressor();
+    Result<InputStream, Void> executeCompress = null;
+    if (compressor != null) {
+      executeCompress = compressPayload(loggerConfig, payloadString);
+      try {
+        payloadString = Base64.encodeBytes(convertToByteArray(executeCompress.getOutput(), streamingHelper));
```
Comment (manikmagar) [inline src/main/java/com/avioconsulting/mule/logger/internal/utils/PayloadTransformer.java:59]: Maybe we can defer this encoding to be done after encryption. Looks like `payloadString` isn't used in `encryptPayload` call if `executeCompress` is provided. Which means this is an unused transformation in that case.

The `encryptPayload` method is doing this same transformation at the end. We could return Result object from encryptPayload method and then do the transformation once at the end of this method. The Result variable could be the same carried from compress method, so whatever is final result, we encode that just once.

File: src/test/java/com/avioconsulting/mule/logger/internal/CustomLoggerOperationTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (manikmagar) [reply src/test/java/com/avioconsulting/mule/logger/internal/CustomLoggerOperationTest.java:97]: @adesjardin @tannersherman Tests are updated in latest commit - https://github.com/avioconsulting/mule-custom-logger/pull/66/commits/114be276b8c98e66868021fbbbd0b8bb47aad8b6 to validate the executions.

**Review by tannersherman — APPROVED** [review]

**Comment by tannersherman** [comment]
Looks good to me after @manikmagar 's tests that he added. But since I also had commits in this PR, @adesjardin's review is also necessary

**Review by adesjardin — APPROVED** [review]

**Pull request merged**